cli/azd/pkg/tools/github/github.go

```diff
@@ -101,7 +98,10 @@ func newGitHubCliImplementation(
 		path:          githubCliPath,
 		commandRunner: commandRunner,
 	}
-
+	if os.Getenv("CODESPACES") == "true" {
```

Comment:
Just wondering - in `git.go` we did this check (and removed the value, if needed) in `newRunArgs`, which feels a little nicer to me (since we don't have to worry about per object state).  Could we move this there?  I think that would also allow us to just remove `forceConfigureAuth` entirely given the other cleanup you've done here?